scripts/update_copyright_headers.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Script to insert legal header to files across the repo
+"""
+
+import glob
+import os
+import traceback
+
+from typing import List
+from typing import Optional
+
+REPO_ROOT_PATH = "../"
+
+EXCLUDE_DIR_NAMES = [
+    "data",
+    "build",
+    "node_modules",
+    "__pycache__",
+]
+
+COPYRIGHT_LINES = [
+    "Copyright (c) Meta Platforms and its affiliates.",
+    "This source code is licensed under the MIT license found in the",
+    "LICENSE file in the root directory of this source tree.",
+]
+
+
+class ProcessingError(Exception):
+    pass
+
+
+class UnsupportedFile(Exception):
+    pass
+
+
+def _add_prefix_suffix(prefix: str, lines: List[str], suffix: Optional[str] = None) -> List[str]:
+    lines = lines[:]
+    for i, line in enumerate(lines):
+        lines[i] = (prefix or "") + line + (suffix or "")
+    return lines
+
+
+def _make_copyright_lines(ext: str) -> List[str]:
+    """
+    Insert copyright as comment lines specific to file extension
+    """
+    lines = []
+
+    ext = ext.lstrip(".")
+    if ext in ["py", "sh", "yml", "yaml"]:
+        lines = _add_prefix_suffix("# ", COPYRIGHT_LINES)
+    elif ext in ["js", "jsx", "ts", "tsx", "css", "scss"]:
+        lines = ["/*"] + _add_prefix_suffix(" * ", COPYRIGHT_LINES) + [" */"]
+    elif ext in ["md", "html"]:
+        lines = ["<!---"] + _add_prefix_suffix("  ", COPYRIGHT_LINES) + ["-->"]
+    else:
+        raise UnsupportedFile(f"Unsupported file extension `{ext}`")
+
+    return _add_prefix_suffix(None, lines, "\n")
+
+
+def _update_copyright_header(file_path: str, replace_existing: bool = False):
+    """
+    Add or replace copyright notice at the top of a file
+    """
+    ext = os.path.splitext(file_path)[1].lower()
+
+    EXAMINED_LINES = 10
+
+    with open(file_path, "r") as f:
+        lines = f.readlines()
+
+    if len(lines) < 2:
+        raise UnsupportedFile("File has fewer than one line")
+
+    # Check copyright presence at the top of the file
+    anchor_line_number = -2
+    for i, line in enumerate(lines[:EXAMINED_LINES]):
+        text = line[:50].lower()
+        if "copyright" in text:
+            anchor_line_number = -1
+        if anchor_line_number == -1 and ("meta" in text or "facebook" in text):
```

Comment:
Could it be `"copyright" in text` instead  of `anchor_line_number == -1` and removing the above line?